src/body.js

```diff
@@ -113,11 +115,20 @@ export default class Body {
 	/**
 	 * Decode response as json
 	 *
-	 * @return  Promise
+	 * @returns {Promise<unknown>}
 	 */
 	async json() {
-		const buffer = await consumeBody(this);
-		return JSON.parse(buffer.toString());
+		const text = await this.text();
+		if (text.length > 50000 && typeof Worker === 'function') {
```

Comment:
don't know why I'm bothered by 50000. Prefer 2 to the power of x such as 

```
2<<15
65535
0xFFFF
0b10000000000000000
```
it is the maximum number representable by 2 bytes and can be defined by the type Uint16
calculating base10 is harder for computers

----

Damaged by all binary knowledge i got and knows how to count and gets easily bothered when someone mistakenly use KB wrongs when they might have meant to use the term `KiB` instead of `KB`

Reply:
Old school? Me too :smile: But I was recently surprised by discovering that KB is officially power of 10 for quite a long time (see http://simplicitysimplifiedfurther.blogspot.com/2013/12/difference-between-kb-and-kib-mb-and.html) and we should resort to `KiB` for the power of 2. But it's just a threshold, we may put whatever number here.

Reply:
Yea, i'm old school i have even had 2 binary watches - one stationary and one arm wrist

<img src="https://www.getdigital.eu/web/getdigital/gfx/products/__generated__resized/380x380/149Binaeruhr_02.jpg">

Reply:
So, 32768 here? I believe most people these days will say such numbers degrade maintainability.

Reply:
Remember that even `.reduce` is blocked by `xo` and `++i` is blocked by AirBnB code style as "hard to understand" 😖

Reply:
32768? Woot?
It shows that the time is 01:03:54

it displays as hh:mm:ss

reading from left to right - bottom to top

--

reduce i can understand `i++` or `++i` is not hard to understand. But like i mention before - hate linters opinion rules - just want a linter that can spot errors.

Reply:
I mean 32768 instead of 50000

Reply:
oh, my bad - nah go full max 16bit value :)
65535 || 0xFFFF

Reply:
I love how `0o100000` looks 😉